sdk/provisioning/Azure.Provisioning/src/Expressions/BicepTypeMapping.cs

```diff
@@ -0,0 +1,156 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using System.Reflection;
+using System.Runtime.Serialization;
+using Azure.Core;
+using Azure.Provisioning.Primitives;
+
+namespace Azure.Provisioning.Expressions;
+
+/// <summary>
+/// Collect the primary logic for mapping .NET types to Bicep types in one place.
+/// </summary>
+internal static class BicepTypeMapping
+{
+    /// <summary>
+    /// Map standard Azure types into Bicep primitive type names like bool,
+    /// int, string, object, or array.  More complex types are not supported.
+    /// </summary>
+    /// <param name="type">A .NET type.</param>
+    /// <returns>A corresponding Bicep type name or null.</returns>
+    public static string? GetBicepTypeName(Type type) =>
+        type == typeof(bool) ? "bool" :
+        type == typeof(int) ? "int" :
+        type == typeof(string) ? "string" :
+        type == typeof(object) ? "object" :
+        type == typeof(Uri) ? "string" :
+        type == typeof(DateTimeOffset) ? "string" :
+        type == typeof(TimeSpan) ? "string" :
+        type == typeof(Guid) ? "string" :
+        type == typeof(IPAddress) ? "string" :
+        type == typeof(ETag) ? "string" :
+        type == typeof(ResourceIdentifier) ? "string" :
+        type == typeof(ResourceType) ? "string" :
+        type == typeof(AzureLocation) ? "string" :
+        type.IsSubclassOf(typeof(Enum)) ? "string" :
+        type.IsSubclassOf(typeof(System.Collections.IEnumerable)) ? "array" :
+        type.IsSubclassOf(typeof(System.Collections.IDictionary)) ? "object" :
+        null;
+
```

Comment:
We also don't expose all of Bicep's type system.  I think we can make it pretty far with primitives and strongly typed objects.  Here's the mapping of everything Azure needs (today) into Bicep primitives.